sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -298,15 +302,28 @@ class SessionCatalog(
         makeQualifiedPath(tableDefinition.storage.locationUri.get)
       tableDefinition.copy(
         storage = tableDefinition.storage.copy(locationUri = Some(qualifiedTableLocation)),
-        identifier = TableIdentifier(table, Some(db)))
+        identifier = tableIdentifier)
     } else {
-      tableDefinition.copy(identifier = TableIdentifier(table, Some(db)))
+      tableDefinition.copy(identifier = tableIdentifier)
     }
 
     requireDbExists(db)
     externalCatalog.createTable(newTableDefinition, ignoreIfExists)
   }
 
+  def validateTableLocation(table: CatalogTable, tableIdentifier: TableIdentifier): Unit = {
+    // SPARK-19724: the default location of a managed table should be non-existent or empty.
+   if (table.tableType == CatalogTableType.MANAGED) {
+     val tableLocation = new Path(defaultTablePath(tableIdentifier))
```

Comment:
We always check `defaultTablePath `? Is that possible, the table location points to the different location?

Reply:
I think managed tables are always created on default path. So the check here should be correct.